thirdeye-spi/src/main/java/ai/startree/thirdeye/spi/datalayer/dto/DatasetConfigDTO.java

```diff
@@ -41,6 +41,15 @@ public class DatasetConfigDTO extends AbstractDTO {
   private Set<String> owners;
   private Boolean active;
   private List<MetricConfigDTO> metrics;
+  /**
+   * Maintain a map of (granularity, timeBucketColumnName).
+   * The timeBucketColumnName column should contain time buckets in milliseconds that
+   * correspond to the granularity.
+   * This info is used to optimize generated queries.
+   * For instance
+   * {"PT5M": "timeBucket5min", "PT1H": "timeBucket1hour"}
+   */
+  private Map<String, String> granularityToBucketTimeColumn;
```

Comment:
What do you think about this?
```  
private Map<Duration, String> granularityToBucketTimeColumn;
```
I'm trying to see if the schema can enforce the validation.

Reply:
I see your point. yeah. time can be in different granularities and this can expand. I think having a container will be better for the long term. we can add complexity later.

Reply:
back to
> I'm trying to see if the schema can enforce the validation.

I saw there is jackson plugin for joda types. I think with this we can introduce schema enforcement with joda types. 
https://github.com/FasterXML/jackson-datatype-joda
I'll try this in an other PR.